js/game.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+function Game(){
+  this.MAX_FRAMES = 10;
+  this.completion = false;
+  this.currentFrame = new Frame();
+  this.frames = [];
+  this.currentFrameNo = 1;
+};
+
+Game.prototype.roll = function(pins) {
+  this._giveBonusPointsToEligibleFrames(pins);
+  this._rollToCurrentFrameIfEligible(pins);
+  this._makeFinalBonusRollIfRequired(pins);
+  this._saveCurrentFrameIfRequired();
+  this._startNewFrameIfRequired();
+  this._completeTheGameIfRequired();
+};
+
+Game.prototype.getScore = function() {
+  return this.frames.reduce(this._sumFrameScores, 0);
+};
+
+Game.prototype.isComplete = function() {
+  return this.completion;
```

Comment:
same thought here about not needing this property and using `_completeTheGameIfRequired()`